scripts/EGAP/create_EGAP_json.py

```diff
@@ -0,0 +1,263 @@
+    """Format the question's response to go in the registration_metadata
+    :param header_row: Header row in spreadsheet
+    :param row: Row in spreadsheet
+    :param question_key: string, Official question key as part of schema
+    :param column_title: string, Corresponding question_key column title in EGAP spreadsheet
+    """
+    index = header_row.index(column_title)
+    value = clean_value(row[index])
+    # Spreadsheet has these as comma-separated values, but looking for array
+    if question_key in ['q33', 'q34']:
+        value = value.split(', ')
+    return build_nested_response(value)
+
+def clean_value(value):
+    """Clean spreadsheet values of issues that will affect validation """
+    if value == 'n/a':
+        return 'N/A'
+    elif value == 'Design was registered before field was added':
+        return ''
+    return value
+
+def build_nested_response(value):
+    return {
+        'comments': [],
+        'extra': [],
+        'value': value
+    }
+
+def validate_response(qid, value):
+    """Validate question response
+
+    Validating each question response individually.  If there is an error, we will
+    attempt to add the value to the corresponding "Other" block.  Return that question id instead.
+
+    For example, q6 is a multiple choice question, with "Other" as a choice.  If text is entered
+    for q6 that does not match one of the multiple choice answers, assuming that this is "other"
+    text, and this response should go to the corresponding q7 question.  q6 will be marked
+    as "Other"
+
+    :param qid: string, question id from schema
+    :param value: question response
+    :param draft: DraftRegistration
+    :return qid: tuple, (qid corresponding to value, optional "Other" qid)
+    """
+    temporary_check = {}
+    temporary_check[qid] = value
+    egap_schema = ensure_schema_structure(from_json('egap-registration.json'))
+    schema = create_jsonschema_from_metaschema(egap_schema,
+        required_fields=False,
+        is_reviewer=False)
+
+    try:
+        json_schema = jsonschema.validate(temporary_check, schema)
+    except ValidationError as exc:
+        if qid in other_mapping:
+            return other_mapping[qid], qid
+        else:
+            raise Exception(exc)
+    return qid, None
+
+def main(default_args=False):
+    if default_args:
+        args = parser.parse_args(['--source', 'default', '--target', 'default'])
+    else:
+        args = parser.parse_args()
+
+    author_source = args.authorsource
+    registry_source = args.registrysource
+    target_directory = args.target
+    dry_run = args.dry
+
+    if not author_source:
+        author_source = 'EGAP_author_emails.csv'
+
+    if not registry_source:
+        registry_source = 'EGAP_registry_for_OSF.csv'
+
+    if not target_directory:
+        target_directory = 'EGAP_data_{}'.format(datetime.datetime.now().strftime('%m-%d-%Y'))
+
+    print author_source
```

Comment:
Nbd deal, but if you could use parenthesis here it would save Python 3 merge conflicts.